client.js

```diff
@@ -1,22 +1,47 @@
-// Example: client.js
 const root = require('./generated/containerPb.js');
 const Container = root.DBMessaging.Protobuf.Container;
 
 class Client {
-  constructor(host, port = 17000, autoReconnect = true) {
+  /**
+   * @param {string} endpoint - The logger endpoint (e.g. "127.0.0.1:17000" or "ws://127.0.0.1:17000")
+   * @param {boolean} [autoReconnect=true] - Automatically reconnect if the connection is lost
+   */
+  constructor(endpoint, autoReconnect = true) {
+    // If endpoint does not start with "ws://" or "wss://", prepend "ws://"
+    let url = endpoint;
+    if (!/^wss?:\/\//.test(url)) {
+      url = `ws://${url}`;
+    }
+
     this.reqId = -1;
     this.autoReconnect = autoReconnect;
+
+    // Time synchronization is enabled by default.
+    this.enableTimeSync = true;
+
     this.isOpen = false;
     this.queuedRequests = {};
     this.storedPromises = {};
     this.nameToId = {};
     this.idToName = {};
+
+    // Time-diff related
     this.timeDiff = 0;
     this.timeReceived = null;
     this.lastTimeRequest = Date.now() / 1000;
     this.haveSentQueuedReq = false;
     this.roundTripTimes = {};
-    this.ws = this._connect(`ws://${host}:${port}`);
+
+    // Create the WebSocket connection
+    this.ws = this._connect(url);
+  }
+
+  /**
+   * Enable or disable time synchronization.
+   * @param {boolean} enable - If true, time sync is enabled; if false, time sync is disabled.
+   */
+  setEnableTimeSync(enable) {
+    this.enableTimeSync = enable;
```

Comment:
The `enableTimeSync` sounds a bit like a method name rather than boolean.

I'd call them:
* field - `timeSyncEnabled`
* getter - `isTimeSyncEnabled()`
* setter - `setTimeSyncEnabled(enable)`